### DISCUSSION TITLE
Snapshot sequence / order

### DISCUSSION THREAD

**Discussion by jojogib** [RomanKubiak/ctrlr#568] [Q&A]
Hey everyone!
How do I change the sequence of a snapshot.
Is it by name like modulator-1 modulator-2 or by vst-index? 
I need to send some modulators before others.
Thank you

**Comment by dnaldoog** [comment]
I have never used snapshot send, but you could make a table of all the modulators you want to send and then loop through that table sending them (possibly using a timer) on panel boot or by clicking on a button etc.

**Comment by damiensellier** [comment]
The "Send Snapshots" function is related to CtrlrPanelMIDISnapshot.cpp :

https://github.com/RomanKubiak/ctrlr/blob/8aa00d82127acda42ad9ac9b7b479461e9436aa4/Source/Core/CtrlrPanel/CtrlrPanelMIDISnapshot.cpp#L48

```
void CtrlrPanelMIDISnapshot::gatherSnapshotData()
{
	buffer.clear();

	for (int i=0; i<owner.getModulators().size(); i++)
	{
		CtrlrModulator *m = owner.getModulators()[i];
		if (m->getMidiMessagePtr())
		{
			if (m->getComponent())
			{
				if ((int)m->getComponent()->getProperty(Ids::componentRadioGroupId) > 0)
				{
					if (m->getComponent()->getToggleState() == false)
						continue;
					else if (!(bool)m->getProperty(Ids::modulatorExcludeFromSnapshot))
						addCtrlrMidiMessageToBuffer (buffer, m->getMidiMessage());
				}
				else if (!(bool)m->getProperty(Ids::modulatorExcludeFromSnapshot))
				{
					addCtrlrMidiMessageToBuffer (buffer, m->getMidiMessage());
				}
			}
		}
	}
```


getModulators() is sorted by modulator Index number. 

```
const Array <CtrlrModulator*> &getModulators()															{ return (ctrlrModulators); }

		/** Get a reference to all modulators with a specified MIDI type. MIDI type is
			determined based on the CtrlrMidiMessage inside the Modulator

			@param typeToFilter					MIDI type to filter by
			@see getModulatorsByUIType
		*/
```


https://github.com/RomanKubiak/ctrlr/blob/8aa00d82127acda42ad9ac9b7b479461e9436aa4/Source/Core/CtrlrModulator/CtrlrModulator.cpp#L24

```
CtrlrModulator::CtrlrModulator(CtrlrPanel &_owner, const int suggestedVstIndex)
	:	ctrlrComponent(0),
		owner(_owner),
		ctrlrModulatorReference(0),
		modulatorTree(Ids::modulator),
		restoreStateStatus(false),
		delayedProperties(Ids::modulator),
		processor(*this),
		vstExported(true)

```


So I guess it sorts in ascending order of vst index.

**Comment by jojogib** [comment]
Thank you. I will try it
